### PR TITLE
Write certificate to local file

### PR DESCRIPTION
**Pull request by pingsutw**
# TL;DR
Webhook and propeller will expose 8080 (Prometheus metrics listener) by default, which causes conflict in the single binary mode. 
- Make `controller manager` externally callable
- Make `ServicePort` configurable
- Make `Init cert` externally callable
- Write cert to local

## Type
 - [ ] Bug Fix
 - [x] Feature
 - [ ] Plugin

## Are all requirements met?

 - [ ] Code completed
 - [ ] Smoke tested
 - [ ] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
image

## Tracking Issue
https://github.com/flyteorg/flyte/issues/2266

fixes https://github.com/flyteorg/flyte/issues/<number>

## Follow-up issue
_NA_

### CODE REVIEW

File: cmd/controller/cmd/root.go [hunk 1]
```diff
@@ -103,12 +105,38 @@ func executeRootCmd(baseCtx context.Context, cfg *config2.Config) error {
...
+	g.Go(func() error {
+		err := profutils.StartProfilingServerWithDefaultHandlers(childCtx, cfg.ProfilerPort.Port, nil)
+		if err != nil {
+			logger.Fatalf(childCtx, "Failed to Start profiling and metrics server. Error: %v", err)
+		}
+		return err
+	})
+
+	g.Go(func() error {
+		err := controller.StartControllerManager(childCtx, mgr)
```
Comment (kumare3) [inline cmd/controller/cmd/root.go:126]: do we have to start this in a go-routine. Earlier we were not doing so?

File: pkg/controller/controller.go [hunk 2]
```diff
@@ -493,8 +493,38 @@ func SharedInformerOptions(cfg *config.Config, defaultNamespace string) []inform
...
+		SyncPeriod:    &cfg.DownstreamEval.Duration,
+		ClientBuilder: executors.NewFallbackClientBuilder((*scope).NewSubScope("kube")),
+	})
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to initialize controller-runtime manager")
+	}
+	return &mgr, nil
+}
+
+func StartControllerManager(ctx context.Context, mgr *manager.Manager) error {
```
Comment (kumare3) [inline pkg/controller/controller.go:519]: lets put some documentation on this

File: cmd/controller/cmd/root.go (hunk 1 above)
Comment (pingsutw) [reply cmd/controller/cmd/root.go:126]: We also started it in go-routine before. https://github.com/flyteorg/flytepropeller/pull/421/files#diff-243ebed2765f75e6a54f57167212fefb08c3b2a85967ad2acbc0eb78919019c1L556-L564

File: pkg/controller/controller.go (hunk 2 above)
Comment (pingsutw) [reply pkg/controller/controller.go:523]: done

**Review by kumare3 — APPROVED** [review]
